### PR TITLE
feat: Ask for confirmation when adding a cluster

### LINKED ISSUE #6159: Be more transparent when adding new clusters via CLI
# Summary

`argocd cluster add` should give the user an indication that it will create a service account with full cluster admin privileges on the remote cluster, and possibly ask for confirmation.

# Motivation

Since this behavior could be unexpected by some, and might break some organisational rules, we should provide users with the required transparency and a possibility to abort.

# Proposal

`argocd cluster add` should print out something like the following and ask the user for confirmation:

```shell
$ argocd cluster add some-context
WARNING: This will create a service account `argocd-manager` on the cluster referenced by context `some-context` with full cluster level admin privileges. Do you want to continue [y/N]? _
```

For automation purposes, the query can be skipped if

* command line flag `--yes` is given to `cluster add` command
* the execution context doesn't have a TTY (to not break existing automation scripts)

### PR DESCRIPTION
**Pull request by dominikbraun**
Closes #6159.

### CODE REVIEW

**Review by jannfis — CHANGES_REQUESTED** [review]
Hi @dominikbraun, thanks a lot for your contribution!

I have a small request for change, please see below.

File: cmd/argocd/commands/cluster.go [hunk 1]
```diff
@@ -76,6 +78,16 @@ func NewClusterAddCommand(clientOpts *argocdclient.ClientOptions, pathOpts *clie
 				log.Fatalf("Context %s does not exist in kubeconfig", contextName)
 			}
 
+			fileInfo, _ := os.Stdout.Stat()
+			isTTY := fileInfo.Mode()&os.ModeCharDevice != 0
```
[the remark this answers is not in the document]
Comment (dominikbraun) [reply cmd/argocd/commands/cluster.go:82]: Thanks! I didn't know this existing function.

File: cmd/argocd/commands/cluster.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package commands
 import (
 	"context"
 	"fmt"
+	"github.com/mattn/go-isatty"
```
Comment (jannfis) [inline cmd/argocd/commands/cluster.go:6]: The linter is complaining about the location of this import. The imports are divided in three distinct blocks, each of which must be alphabetically sorted:

1. Golang standard libary
2. Own package imports
3. External package imports

Can you please move the import line to its own block to quiet the linter?

**Review by jannfis — COMMENTED** [review]
Can you please fix the linter issue as described below?

Also, our CLI docs are auto-generated. Please run `make clidocsgen` and commit the changes, so that the `codegen` CI check will succeed.

Thanks!

**Comment by dominikbraun** [comment]
Thanks, I generated the CLI docs.

**Review by jannfis — APPROVED** [review]
LGTM. thank you @dominikbraun !

**Pull request merged**